bigquery_etl/cli/metadata.py

```diff
@@ -94,10 +98,41 @@ def update(name: str, sql_dir: Optional[str], project_id: Optional[str]) -> None
             dataset_metadata_updated = True
         else:
             if table_metadata.workgroup_access is None:
-                table_metadata.workgroup_access = (
-                    dataset_metadata.default_table_workgroup_access
-                )
-                table_metadata_updated = True
+                table_metadata.workgroup_access = []
+
+            for (
+                default_workgroup_access
+            ) in dataset_metadata.default_table_workgroup_access:
+                role_exists = False
+                for i, table_workgroup_access in enumerate(
+                    table_metadata.workgroup_access
+                ):
+                    if table_workgroup_access.role == default_workgroup_access.get(
+                        "role"
+                    ):
+                        role_exists = True
+                        table_metadata_members = table_workgroup_access.members
+                        table_metadata_members.extend(
+                            default_workgroup_member
+                            for default_workgroup_member in default_workgroup_access.get(
+                                "members", []
+                            )
+                            if default_workgroup_member not in table_metadata_members
+                        )
+
+                        table_metadata.workgroup_access[i].members = (
+                            table_metadata_members
+                        )
```

Comment:
A bit code golfy but I think this is equivalent and would be easier to read
```suggestion
                        table_metadata.workgroup_access[i].members = sorted(
                            set(table_workgroup_access.members)
                            | set(default_workgroup_access.get("members", []))
                        )
```